lib/host_remove_duplicates.py

```diff
@@ -0,0 +1,166 @@
+from sqlalchemy import and_
+from sqlalchemy import not_
+from sqlalchemy import or_
+
+from app.models import Host
+from lib.metrics import delete_duplicate_host_count
+
+# from app.logging import get_logger
+
+# logger = None
+
+__all__ = ("delete_duplicate_hosts",)
+
+
+# The order is important, particularly the first 3 which are elevated facts with provider_id being the highest priority
+CANONICAL_FACTS = ("fqdn", "satellite_id", "bios_uuid", "ip_addresses", "mac_addresses")
+
+ELEVATED_CANONICAL_FACT_FIELDS = ("provider_id", "insights_id", "subscription_manager_id")
+
+
+def matches_at_least_one_canonical_fact_filter(canonical_facts):
+    # Contains at least one correct CF value
+    # Correct value = contains key:value
+    # -> OR( *correct values )
+    filter_ = ()
+    for key, value in canonical_facts.items():
+        filter_ += (Host.canonical_facts.contains({key: value}),)
+
+    return or_(*filter_)
```

Comment:
i recal lthat this should do the job as well
```suggestion
    return or_(Host.canonical_facts.contains({key: value}) for key, value in canonical_facts.items())
```